app/src/lib/stores/app-store.ts

```diff
@@ -1852,6 +1852,38 @@ export class AppStore extends TypedBaseStore<IAppState> {
     this._updateCurrentPullRequest(repository)
     this.updateMenuItemLabels(repository)
     this._initializeCompare(repository)
+    this.refreshRepositoryState([repository])
+  }
+
+  public refreshAllRepositories() {
+    return this.refreshRepositoryState(this.repositories)
+  }
+
+  private async refreshRepositoryState(
+    repositories: ReadonlyArray<Repository>
+  ): Promise<void> {
+    if (!enableRepoInfoIndicators()) {
+      return
+    }
+
+    for (const repo of repositories) {
+      await this.withAuthenticatingUser(repo, async (repo, account) => {
```

Comment:
More of a question than a suggestion. Could we do this in parallel? It looks like nothing is modifying any share state except for L#1878.

Reply:
I am not sure I follow -- could you please elaborate on how you'd do this in parallel?

Reply:
Ah I now see what you mean - I like that!

Reply:
@vanessayuenn @iAmWillShepherd I don't have strong feelings on either approach, but I felt serially executing these refresh operations felt more suited to this background work. Just needed to ensure you `await`-ed the `this.withAuthenticatingUser` to not create fire-and-forget promises.